lib/ansible/parsing/yaml/constructor.py

```diff
@@ -44,6 +44,8 @@ def construct_yaml_map(self, node):
         value = self.construct_mapping(node)
         data.update(value)
         data.ansible_pos = self._node_position_info(node)
+        if hasattr(value, 'merge_behavior'):
+            data.merge_behavior = value.merge_behavior
```

Comment:
I think `merge_behavior` should be a proper native attribute on `AnsibleMapping` or `AnsibleBaseYAMLObject`, and supported by things like `update()` so that it's automatically copied instead of having to be done manually like this.